### PR TITLE
Fix bugs and typos in CP Power Iteration

### PR DESCRIPTION
**Pull request by chrisyeh96**
and remove unused import

### CODE REVIEW

File: tensorly/decomposition/_symmetric_cp.py [hunk 1]
```diff
@@ -33,30 +32,30 @@ def symmetric_power_iteration(tensor, n_repeat=10, n_iteration=10, verbose=False
...
     order = tl.ndim(tensor)
     size = tl.shape(tensor)[0]
-    
+
     if not tl.shape(tensor) == (size, )*order:
         raise ValueError('The input tensor does not have the same size along each mode.')
 
     # A list of candidates for each mode
-    best_score = 0
+    best_score = -np.inf
```
Comment (JeanKossaifi) [inline tensorly/decomposition/_symmetric_cp.py:40]: Best not to mix numpy in the calculations, let's just not initialize best_score here.

File: tensorly/decomposition/_symmetric_cp.py [hunk 2]
```diff
@@ -33,30 +32,30 @@ def symmetric_power_iteration(tensor, n_repeat=10, n_iteration=10, verbose=False
...
     scores = []
     modes = list(range(1, order))
-    
+
     for _ in range(n_repeat):
```
Comment (JeanKossaifi) [inline tensorly/decomposition/_symmetric_cp.py:44]:
```suggestion
    for i in range(n_repeat):
```

File: tensorly/decomposition/_symmetric_cp.py [hunk 3]
```diff
@@ -33,30 +32,30 @@ def symmetric_power_iteration(tensor, n_repeat=10, n_iteration=10, verbose=False
...
         for _ in range(n_iteration):
             for _ in range(order):
                 factor = tl.tenalg.multi_mode_dot(tensor, [factor]*(order-1), modes=modes)
                 factor = factor / tl.norm(factor, 2)
-                
+
         score = tl.tenalg.multi_mode_dot(tensor, [factor]*order)
         scores.append(score) #round(score, 2))
         
         if score > best_score:
```
Comment (JeanKossaifi) [inline tensorly/decomposition/_symmetric_cp.py:55]:
```suggestion
        if (i == 0) or (score > best_score):
```

**Review by JeanKossaifi — CHANGES_REQUESTED** [review]
Thanks @chrisyeh96, looks good, just made some suggestions to avoid using `np.inf`.
After this I think it's ready to merge.

**Comment by chrisyeh96** [comment]
@JeanKossaifi

I made the proposed fixes. I also made the same fixes in the general `power_iteration()` method (for non-symmetric tensors).

**Comment by JeanKossaifi** [comment]
@chrisyeh96 this is great, thank you! Merging.

**Pull request merged**